intermol/desmond/desmond_parser.py

```diff
@@ -1025,32 +1026,42 @@ def loadMAtoms(self, lines, start, end, currentMolecule, slength):
                     if 'r_ffio_z_vel' in cols:
                         atom.velocity[2] = float(aline[cols['r_ffio_z_vel']]) * units.angstroms * units.picoseconds**(-1)
 
+                    pdbname = aname = None
                     if 's_m_pdb_atom_name' in cols:
                         pdbaname = aline[cols['s_m_pdb_atom_name']].strip()
                     if 's_m_atom_name' in cols:
                         aname = aline[cols['s_m_atom_name']].strip()
-                    if re.match('$^',pdbaname) and not re.match('$^',aname):
-                        atom.name = aname
-                    elif re.match('$^',aname) and not re.match('$^',pdbaname):
-                        atom.name = pdbaname
-                    elif re.search("\d+",pdbaname) and not re.search("\d+",aname):
-                        if re.search("\D+",pdbaname) and re.search("\w+",pdbaname):
-                            atom.name = pdbaname
-                        else:
-                            atom.name = aname
-                    elif re.search("\d+",aname) and not re.search("\d+",pdbaname):
-                        if re.search("\D+",aname) and re.search("\w+",aname):
+
+                    if pdbname and aname:
+                        if re.match('$^', pdbaname) and not re.match('$^', aname):
                             atom.name = aname
-                        else:
+                        elif re.match('$^', aname) and not re.match('$^', pdbaname):
                             atom.name = pdbaname
-                    elif re.match('$^',pdbaname) and re.match('$^',aname):
-                        atom.name = "None"
+                        elif re.search("\d+", pdbaname) and not re.search("\d+", aname):
+                            if re.search("\D+", pdbaname) and re.search("\w+", pdbaname):
+                                atom.name = pdbaname
+                            else:
+                                atom.name = aname
+                        elif re.search("\d+", aname) and not re.search("\d+", pdbaname):
+                            if re.search("\D+", aname) and re.search("\w+", aname):
+                                atom.name = aname
+                            else:
+                                atom.name = pdbaname
+                        elif re.match('$^', pdbaname) and re.match('$^', aname):
+                            atom.name = "None"
+                        else:
+                            atom.name = aname  #doesn't matter which we choose, so we'll go with atom name instead of pdb
```

Comment:
@mrshirts could you explain what exactly the intended function of the regex chunk below is? I suspect it could be rewritten much more cleanly and probably without regex.